### DISCUSSION TITLE
Get HTTP request headers

### DISCUSSION THREAD

**Discussion by subesokun** [remix-run/remix#4384] [Q&A] [answered]
Hi, I'm running a remix app behind a middleware/proxy that performs the user authentication via SSO. It adds some extra headers to the HTTP request and now I'm wondering how I can access those headers to extract the relevant information like the user ID? Any help will be appreciated

**Comment by lili21** [comment, marked as answer]
you can get that from `request.headers.get('xx')` inside `loader` or `action`

```js
export const loader = async ({ request }) => {
  const id = request.headers.get('id') // 
}
```

**Reply by subesokun** [reply]
Awesome, thanks a lot!
